applications/GeoMechanicsApplication/custom_workflows/dgeoflow.cpp

```diff
@@ -253,61 +237,66 @@ namespace Kratos
+                                     const GeoMechanicsNewtonRaphsonErosionProcessStrategyType::Pointer& rpSolvingStrategy,
+                                     double                                                              Time,
+                                     double                                                              DeltaTime,
+                                     unsigned int                                                        NumberOfIterations)
     {
 
     	// Initialize
-        for (auto process : processes)
+        for (const auto& process : rProcesses)
         {
             process->ExecuteInitialize();
         }
 
-        for (auto process : processes)
+        for (const auto& process : rProcesses)
         {
             process->ExecuteBeforeSolutionLoop();
         }
 
-        for (unsigned int iter = 0; iter < number_iterations; ++iter)
+        for (unsigned int iter = 0; iter < NumberOfIterations; ++iter)
         {
-            time += delta_time;
-            model_part.CloneTimeStep(time);
-            p_solving_strategy->Initialize();
-            p_solving_strategy->InitializeSolutionStep();
+            Time += DeltaTime;
+            rModelPart.CloneTimeStep(Time);
+            rpSolvingStrategy->Initialize();
+            rpSolvingStrategy->InitializeSolutionStep();
 
-            for (auto process : processes)
+            for (const auto& process : rProcesses)
             {
                 process->ExecuteInitializeSolutionStep();
             }
 
-            p_solving_strategy->Predict();
-            p_solving_strategy->SolveSolutionStep();
+            rpSolvingStrategy->Predict();
+            rpSolvingStrategy->SolveSolutionStep();
 
-            for (auto process : processes)
+            for (const auto& process : rProcesses)
             {
                 process->ExecuteFinalizeSolutionStep();
             }
 
-            p_solving_strategy->FinalizeSolutionStep();
+            rpSolvingStrategy->FinalizeSolutionStep();
         }
 
-        for (auto process : processes)
+        for (const auto& process : rProcesses)
         {
             process->ExecuteFinalize();
         }
 
         return 0;
     }
 
-    int KratosExecute::execute_flow_analysis(std::string workingDirectory, std::string projectName,
-                                             double minCriticalHead, double maxCriticalHead, double stepCriticalHead,
-                                             std::string criticalHeadBoundaryModelPartName,
-                                             std::function<void(char *)> logCallback,
-                                             std::function<void(double)> reportProgress,
-                                             std::function<void(char *)> reportTextualProgress,
-                                             std::function<bool()> shouldCancel)
+    int KratosExecute::ExecuteFlowAnalysis(const std::string&                       rWorkingDirectory,
```

Comment:
I recommend std::filesystem::path for this purpose

Reply:
Yes, I totally agree. But let's do that in a separate PR. There are many more things that can be improved here, but I prefer to take small steps. Anyway, your suggestion is greatly appreciated. Thanks.

Reply:
Sure, just wanted to point it out in general